### PR TITLE
Add a check if current range is in a padding element and update it if necessary before executing a command.

### PR DESCRIPTION
**Pull request by yrucrem**
Admin please build this request

### CODE REVIEW

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/301/<h2>Build result: FAILURE</span></h2>[...truncated 580 lines...][INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/aloha/command.js:97:17: Expected '+' at column 21, not column 17.[INFO] 				+ 'determine the corresponding block element');[INFO]                 ^[INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/aloha/command.js:97:62: Expected ')' at column 13, not column 62.[INFO] 				+ 'determine the corresponding block element');[INFO]                                                              ^[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 23:39.535s[INFO] Finished at: Wed Jul 01 12:22:10 CEST 2015[INFO] Final Memory: 81M/2937M[INFO] ------------------------------------------------------------------------Waiting for Jenkins to finish collecting data[ERROR] Failed to execute goal com.googlecode.jslint4java:jslint4java-maven-plugin:2.0.2:lint (lint-core) on project alohaeditor: JSLint found 3 problems in 42 files -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[JENKINS] Archiving /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/pom.xml to org.alohaeditor/alohaeditor/1.1.28-SNAPSHOT/alohaeditor-1.1.28-SNAPSHOT.pomchannel stoppedSetting status of 93d84f985e65fbac1a55f83ba94c87587ff525e1 to FAILURE with url http://jenkins.office/job/alohaeditor-upstream-pullrequests/301/ and message: Build finished.
Test FAILed.

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/305/<h2>Build result: FAILURE</span></h2>[...truncated 583 lines...][INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/aloha/command.js:97:17: Expected '+' at column 21, not column 17.[INFO] 				+ 'determine the corresponding block element');[INFO]                 ^[INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/aloha/command.js:97:62: Expected ')' at column 13, not column 62.[INFO] 				+ 'determine the corresponding block element');[INFO]                                                              ^[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 25:03.008s[INFO] Finished at: Mon Jul 06 17:18:35 CEST 2015[INFO] Final Memory: 83M/2955M[INFO] ------------------------------------------------------------------------Waiting for Jenkins to finish collecting data[ERROR] Failed to execute goal com.googlecode.jslint4java:jslint4java-maven-plugin:2.0.2:lint (lint-core) on project alohaeditor: JSLint found 4 problems in 42 files -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[JENKINS] Archiving /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/pom.xml to org.alohaeditor/alohaeditor/1.1.28-SNAPSHOT/alohaeditor-1.1.28-SNAPSHOT.pomchannel stoppedSetting status of 30b02fcaadfcbd1bc8cbc0f37db826043650a064 to FAILURE with url http://jenkins.office/job/alohaeditor-upstream-pullrequests/305/ and message: Build finished.
Test FAILed.

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/306/<h2>Build result: FAILURE</span></h2>[...truncated 577 lines...][INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/util/misc.js:109:17: Expected '!' at column 21, not column 17.[INFO] 				!jQuery(elem).find('.aloha-block').length) {[INFO]                 ^[INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/aloha/command.js:86:51: Don't use octal: '\0'. Use '\u....' instead.[INFO] 					range.startContainer.data = '\0xa0' + text;[INFO]                                                   ^[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 25:30.609s[INFO] Finished at: Mon Jul 06 17:44:26 CEST 2015[INFO] Final Memory: 86M/2937M[INFO] ------------------------------------------------------------------------Waiting for Jenkins to finish collecting data[ERROR] Failed to execute goal com.googlecode.jslint4java:jslint4java-maven-plugin:2.0.2:lint (lint-core) on project alohaeditor: JSLint found 2 problems in 42 files -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[JENKINS] Archiving /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/pom.xml to org.alohaeditor/alohaeditor/1.1.28-SNAPSHOT/alohaeditor-1.1.28-SNAPSHOT.pomchannel stoppedSetting status of 82ce42e8a10b39220b89dc49bff69c9c43fdf3bb to FAILURE with url http://jenkins.office/job/alohaeditor-upstream-pullrequests/306/ and message: Build finished.
Test FAILed.

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/307/<h2>Build result: FAILURE</span></h2>[...truncated 575 lines...]log4j:WARN Please initialize the log4j system properly.[INFO] [INFO] --- jslint4java-maven-plugin:2.0.2:lint (lint-core) @ alohaeditor ---[INFO] /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/src/lib/util/misc.js:109:17: Expected '!' at column 21, not column 17.[INFO] 				!jQuery(elem).find('.aloha-block').length) {[INFO]                 ^[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 23:04.826s[INFO] Finished at: Mon Jul 06 19:21:29 CEST 2015[INFO] Final Memory: 86M/2937M[INFO] ------------------------------------------------------------------------Waiting for Jenkins to finish collecting data[ERROR] Failed to execute goal com.googlecode.jslint4java:jslint4java-maven-plugin:2.0.2:lint (lint-core) on project alohaeditor: JSLint found 1 problems in 42 files -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[JENKINS] Archiving /opt/jenkins/.jenkins/jobs/alohaeditor-upstream-pullrequests/workspace/pom.xml to org.alohaeditor/alohaeditor/1.1.28-SNAPSHOT/alohaeditor-1.1.28-SNAPSHOT.pomchannel stoppedSetting status of 18db336744f778b7a1a018c97e42452434802203 to FAILURE with url http://jenkins.office/job/alohaeditor-upstream-pullrequests/307/ and message: Build finished.
Test FAILed.

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/308/
Test PASSed.

File: src/lib/aloha/command.js [hunk 1]
```diff
@@ -39,6 +39,66 @@ define([
...
+				&& container.previousSibling.id == paddedId)) {
+			if (range.startOffset == 0) {
+				if (text.substr(0, 1) != '\xa0') {
+					range.startContainer.data = '\xa0' + text;
+				}
+
+				range.setStart(range.startContainer, 1);
+				range.setEnd(range.endContainer, 1);
+			}
+		} else {
```
Comment (npomaroli) [inline src/lib/aloha/command.js]: Remove the else branch

**Comment by GenticsDev** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.office:8080//job/alohaeditor-upstream-pullrequests/309/
Test PASSed.

**Pull request merged**